app/queries/find_latest_matching_subscription.rb

```diff
@@ -0,0 +1,12 @@
+class FindLatestMatchingSubscription
+  def self.call(original_subscription)
+    subscriber_list_id = original_subscription.subscriber_list_id
+    subscriber_id = original_subscription.subscriber_id
+    frequency = original_subscription.frequency
+    latest_subscription = Subscription.
+        where(subscriber_list_id: subscriber_list_id, subscriber_id: subscriber_id, frequency: frequency).
+        order("created_at DESC")
+        .first
+    latest_subscription
```

Comment:
Out of interest, is it possible that there could be no matches? If so, what should we do if this is the case?
Additionally (on a super pedantic note) you could just have the query (ie `Subscription.where...` as the last line and not both copying it into a variable and then returning it but if you have  a good readability reason then I can acquiesce

Reply:
Thanks, have fixed up the return value 👍 

There _should_ always be a match, because we're passing a valid subscription to the method, so when we do the database grab we should at the very least be getting back that same subscription! But to be ultra safe, I could wrap it in a `begin/rescue` and return the original subscription...?

What do you think? Felt to me like we'd be covering a case that will never happen (it's hard to even write a test for that!) but I'm happy to have a go 😄

Reply:
Cool. I was just clarifying - I think it's fine to leave as is, I forgot that subscriptions don't get deleted, just marked as ended. 👍